test/e2e/parallel/proxy_test.go

```diff
@@ -1033,13 +1041,141 @@ func TestSpaceLister(t *testing.T) {
 		require.NoError(t, err)
 	})
 
-	t.Run("banned user cannot create proxy client", func(t *testing.T) {
-		// when
-		_, err := hostAwait.CreateAPIProxyClient(t, users["banneduser"].token, hostAwait.APIProxyURL)
+	t.Run("banned user", func(t *testing.T) {
+		bannedUser := users["banneduser"]
 
-		// then
-		// this is the error expected to be returned by the proxy when the user is banned
-		require.ErrorContains(t, err, "unable to get target cluster: no member cluster found for the user")
+		t.Run("cannot get apis list", func(t *testing.T) {
+			req, err := http.NewRequest(http.MethodGet, fmt.Sprintf("%s%s", hostAwait.APIProxyURL, "/apis"), nil)
+			require.NoError(t, err)
+
+			req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", bannedUser.token))
+			tr := &http.Transport{TLSClientConfig: &tls.Config{InsecureSkipVerify: true}} // nolint:gosec
+			client := &http.Client{Transport: tr}
+			res, err := client.Do(req)
+			require.NoError(t, err)
+			defer res.Body.Close()
+
+			require.Equal(t, http.StatusForbidden, res.StatusCode)
+		})
+
+		t.Run("cannot get workspace", func(t *testing.T) {
+			_, err := bannedUser.getWorkspace(t, hostAwait, bannedUser.compliantUsername)
```

Comment:
Do we really need to use `getWorkspace()` here? It uses a poll (it's slow because it will keep trying until it's timed out) and hides the actual error. Shouldn't we make sure that everything is ready (spaces, space bindings, etc) and then just use the client directly `proxyCl.Get(...)` and then check that the error is `Forbidden` with the expected error message?

Reply:
great point, let me fix that

Reply:
done in a533a2b